zk/conn.go

```diff
@@ -528,48 +600,58 @@ func (c *Conn) authenticate() error {
 	return nil
 }
 
-func (c *Conn) sendLoop(conn net.Conn, closeChan <-chan struct{}) error {
+func (c *Conn) sendData(req *request) error {
+	buf := make([]byte, bufferSize)
```

Comment:
Should move this buffer to Conn instead of allocating for each request. (bufferSize is 1.5MB)